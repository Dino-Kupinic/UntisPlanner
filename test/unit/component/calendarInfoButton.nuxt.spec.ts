import {describe, test, it, expect, vi} from "vitest"
import {mountSuspended} from "@nuxt/test-utils/runtime"
import {CalendarInfoButton} from "#components"

describe("CalendarInfoButton", async () => {
  const component = await mountSuspended(CalendarInfoButton)

  test("it can mount properly", async () => {
    expect(component.text()).toMatchSnapshot()
  })

  it("contains the correct label", async () => {
    expect(component.text()).toBe("Info")
  })

  it("contains a button", async () => {
    expect(component.find("button").exists()).toBeTruthy()
  })

  it.skip("shows a floating window when clicked", async () => {
    await component.find("button").trigger("click")
    // TODO: complete test
  })

  it("has the correct distance and trigger", async () => {
    expect(component.findComponent("VDropdown").attributes()).toStrictEqual({distance: "10", triggers: "click"})
  })
})
