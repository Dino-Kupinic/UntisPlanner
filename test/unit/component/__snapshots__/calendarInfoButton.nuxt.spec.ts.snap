// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CalendarInfoButton > it can mount properly 1`] = `"Info"`;
